subprojects/launcher/src/main/java/org/gradle/launcher/daemon/registry/DaemonRegistryContent.java

```diff
@@ -69,8 +70,8 @@ public DaemonInfo getInfo(Address address) {
     /**
      * Removes the status
      */
-    public void removeInfo(Address address) {
```

Comment:
Instead of just passing the Address, could we also pass the daemon's UID?  I think the client has that information too and it would disambiguate between the two.

Reply:
Actually, this `Address` is `MultiChoiceAddress`, which already contains daemon's UUID. But what's the point? The point is that we need to remove all entries with same port, regardless of UUID.

Reply:
After some poking around I think the change makes sense for now.

`MultiChoiceAddress` is used as a key in a map with daemon infos while its equality depends on both the UUID and the port. In the given case UUIDs are unique, while the ports clash. 
Which makes me wonder (out of scope for this change):
- if it is the port that should uniquely identify the daemon - why not map by port and not by address directly?
- if all addresses are `InetEndpoints` (which looks to be the case), why don't we use `InetEndpoint` and not `Address` as the base interface in all the places and remove the casts.